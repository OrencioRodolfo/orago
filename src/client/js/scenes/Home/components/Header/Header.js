import React from 'react';
import styled from 'styled-components';
import { media } from '../../../../theme/style-utils';
import RestricitedContainer from '../../../../components/Containers/Restricited/Restricited';
import SpaceBwContainer from '../../../../components/Containers/SpaceBetween/SpaceBetween';
import Social from '../Social/Social';

const HeaderWrapper = styled.header`
  background-color: ${props => props.theme.color('foreground')};
  padding: 12px 0;
  color: ${props => props.theme.contrast('dark')};
  ${media.phone`display: none;`}
`;

const Tag = styled.span`
  color: ${props => props.theme.contrast('dark', 'secondary')};
`;

const Header = () => (
  <HeaderWrapper>
    <RestricitedContainer>
      <SpaceBwContainer>
        <Tag>@ogoncalves</Tag>
        <Social />
      </SpaceBwContainer>
    </RestricitedContainer>
  </HeaderWrapper>
);

export default Header;
