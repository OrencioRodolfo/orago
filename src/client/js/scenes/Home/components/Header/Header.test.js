import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

it('Must render the left content of a pill button', () => {
  const wrapper = shallow(
    <Header>
      Left content
    </Header>,
  );
  expect(wrapper).toMatchSnapshot();
});
