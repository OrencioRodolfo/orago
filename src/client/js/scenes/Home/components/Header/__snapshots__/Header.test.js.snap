// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must render the left content of a pill button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header>
    Left content
  </Header>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Container>
        <styled.section>
          <span>
            @orenciorodolfo
          </span>
          <styled.div>
            <Button
              icon="github fa-2x"
            />
            <Button
              icon="linkedin fa-2x"
            />
            <Button
              icon="twitter fa-2x"
            />
            <Button
              icon="facebook fa-2x"
            />
          </styled.div>
        </styled.section>
      </Container>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": <styled.section>
          <span>
            @orenciorodolfo
          </span>
          <styled.div>
            <Button
              icon="github fa-2x"
            />
            <Button
              icon="linkedin fa-2x"
            />
            <Button
              icon="twitter fa-2x"
            />
            <Button
              icon="facebook fa-2x"
            />
          </styled.div>
        </styled.section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <span>
              @orenciorodolfo
            </span>,
            <styled.div>
              <Button
                icon="github fa-2x"
              />
              <Button
                icon="linkedin fa-2x"
              />
              <Button
                icon="twitter fa-2x"
              />
              <Button
                icon="facebook fa-2x"
              />
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "@orenciorodolfo",
            },
            "ref": null,
            "rendered": "@orenciorodolfo",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Button
                  icon="github fa-2x"
                />,
                <Button
                  icon="linkedin fa-2x"
                />,
                <Button
                  icon="twitter fa-2x"
                />,
                <Button
                  icon="facebook fa-2x"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "github fa-2x",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "linkedin fa-2x",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "twitter fa-2x",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "facebook fa-2x",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Container>
          <styled.section>
            <span>
              @orenciorodolfo
            </span>
            <styled.div>
              <Button
                icon="github fa-2x"
              />
              <Button
                icon="linkedin fa-2x"
              />
              <Button
                icon="twitter fa-2x"
              />
              <Button
                icon="facebook fa-2x"
              />
            </styled.div>
          </styled.section>
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <styled.section>
            <span>
              @orenciorodolfo
            </span>
            <styled.div>
              <Button
                icon="github fa-2x"
              />
              <Button
                icon="linkedin fa-2x"
              />
              <Button
                icon="twitter fa-2x"
              />
              <Button
                icon="facebook fa-2x"
              />
            </styled.div>
          </styled.section>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <span>
                @orenciorodolfo
              </span>,
              <styled.div>
                <Button
                  icon="github fa-2x"
                />
                <Button
                  icon="linkedin fa-2x"
                />
                <Button
                  icon="twitter fa-2x"
                />
                <Button
                  icon="facebook fa-2x"
                />
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "@orenciorodolfo",
              },
              "ref": null,
              "rendered": "@orenciorodolfo",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Button
                    icon="github fa-2x"
                  />,
                  <Button
                    icon="linkedin fa-2x"
                  />,
                  <Button
                    icon="twitter fa-2x"
                  />,
                  <Button
                    icon="facebook fa-2x"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "github fa-2x",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "linkedin fa-2x",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "twitter fa-2x",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "facebook fa-2x",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
